src/main/java/com/minecolonies/core/event/EventHandler.java

```diff
@@ -123,9 +124,15 @@ public static void onEntityAdded(@NotNull final EntityJoinLevelEvent event)
               .getType()
               .is(ModTags.mobAttackBlacklist)))
             {
-                ((Mob) event.getEntity()).targetSelector.addGoal(6, new NearestAttackableTargetGoal<>((Mob) event.getEntity(), EntityCitizen.class, true, citizen -> !citizen.isInvisible()));
+                ((Mob) event.getEntity()).targetSelector.addGoal(6,
+                  new NearestAttackableTargetGoal<>((Mob) event.getEntity(), EntityCitizen.class, true, citizen -> !citizen.isInvisible()));
                 ((Mob) event.getEntity()).targetSelector.addGoal(7, new NearestAttackableTargetGoal<>((Mob) event.getEntity(), EntityMercenary.class, true));
             }
+
+            if (event.getEntity() instanceof AbstractFastMinecoloniesEntity && ((ServerLevel) event.getLevel()).getEntity(event.getEntity().getUUID()) != null)
+            {
+                event.setCanceled(true);
```

Comment:
Cancelling the event entirely seems potentially troublesome.  It's still actually adding the entity to the world, and some things might want to be aware of that.  e.g. perhaps mapping mods or entity radars.

Reply:
![image](https://github.com/ldtteam/minecolonies/assets/38401808/c46d093f-432a-443d-9341-9b9cd657ae0d)